### PR TITLE
Use tempy in testCommon

### LINKED ISSUE #177: Refactor use of rimraf in test suite
Not sure how to achieve this, but ideally browser implementations don't need browserify hacks (https://github.com/Level/memdown/issues/114). What's more, the behavior should not be browser-specific. Let the implementation decide whether to use `rimraf`. E.g. memdown in node shouldn't. Currently this can be only achieved by overriding `testCommon` entirely.

Related to #157, cc @ralphtheninja.

### LINKED ISSUE #240: Don't use __dirname in testCommon
https://github.com/Level/abstract-leveldown/blob/d806bf147d12186990db7fdc6da4c95e7a558ccc/testCommon.js#L8

Should probably use `process.cwd()`, or better yet, a unique temporary directory.

Ref https://github.com/Level/leveldown-hyper/pull/68#discussion_r196815629

### LINKED ISSUE #248: Should level-concat-iterator be a dependency?
Otherwise implementations that use the default `testCommon` must include `level-concat-iterator` as a devDependency themselves.

https://github.com/Level/abstract-leveldown/blob/029f56acb2123944045430c54a2fdbdfb69da8d4/package.json#L19-L21

### PR DESCRIPTION
**Pull request by vweevers**
Closes #240, closes #248, closes #177.

### CODE REVIEW

**Comment by vweevers** [comment]
Note regarding removal of `testCommon.cleanup`: `leveldown` and friends use it, but they don't have to anymore after this. See https://github.com/Level/leveldown/issues/491

File: testCommon.js [hunk 1]
```diff
@@ -1,57 +1,13 @@
...
-var setUp = function (t) {
-  cleanup(function (err) {
-    t.error(err, 'cleanup returned an error')
-    t.end()
-  })
-}
-
-var tearDown = function (t) {
-  setUp(t) // same cleanup!
-}
```
Comment (ralphtheninja) [inline testCommon.js]: Loads of red. Very nice!

**Comment by ralphtheninja** [comment]
What happened with the browser tests there?

**Comment by vweevers** [comment]
image

😨

**Comment by vweevers** [comment]
Ah, of course, the browser tests require `testCommon`. Need to add a browserify exclusion for `tempy`.

**Comment by vweevers** [comment]
Better yet, use a custom `testCommon` in browsers.

File: package.json [hunk 2]
```diff
@@ -29,6 +29,9 @@
     "test-browsers": "airtap --sauce-connect --loopback airtap.local test.js",
     "test-browser-local": "airtap --local -- test.js"
   },
+  "browser": {
+    "./testCommon.js": "./test-common-browser.js"
```
Comment (vweevers) [inline package.json]: @ralphtheninja how about renaming `testCommon` to `test-common`?

Comment (ralphtheninja) [reply package.json]: Agreed.

Comment (ralphtheninja) [reply package.json]: Camel case in file names is annoying :)

**Pull request merged**